python_modules/dagster/dagster/_core/storage/event_log/in_memory.py

```diff
@@ -34,9 +34,13 @@ def __init__(self, inst_data: Optional[ConfigurableClassData] = None, preload=No
         # hold one connection for life of instance, but vend new ones for specific calls
         self._held_conn = self._engine.connect()
 
-        SqlEventLogStorageMetadata.create_all(self._held_conn)
-        alembic_config = get_alembic_config(__file__, "sqlite/alembic/alembic.ini")
-        stamp_alembic_rev(alembic_config, self._held_conn)
+        with self._connect() as conn:
+            SqlEventLogStorageMetadata.create_all(conn)
+        with self._connect() as conn:
+            # need to do this as a separate connection since stamp_alembic_rev closes the current
+            # connection
```

Comment:
really? what an odd API - they expect you to open a connection and then they close it for you?

does that mean we are closing it twice - should we not wrap it in a contextmanager then?

Reply:
you're right... it's not being closed... it was locked and this was just pending commits because of the transaction changes.